tests/ixia/test_ixia_traffic_restpy.py

```diff
@@ -0,0 +1,153 @@
+###############################################################################
+# This test cases demonstrates: 
+#   * All the fixtures required for running ixia script (please see the 
+#     arguments of the test function)
+#   * How Ixia chassis card/ports are addressed
+#   * How you can configure/control ixia devices, start traffic and collect 
+#     statistics using REST API
+#   * This simple sanity test cases can be used to check if testbed setup
+#     is correct or not - since it prints a lot of testbed data
+###############################################################################
+
+import logging
+import time
+import pytest
+import ipaddr
+from common.utilities import wait_until
+from common.fixtures.conn_graph_facts import conn_graph_facts, fanout_graph_facts 
+from common.reboot import *
+from ixnetwork_restpy import SessionAssistant, Files
+
+from common.ixia.ixia_fixtures import ixia_api_serv_ip, ixia_api_serv_user,\
+     ixia_api_serv_passwd, ixia_api_serv_port, ixia_api_serv_session_id, \
+     ixia_api_server_session  
+
+from common.ixia.ixia_helpers import get_neigh_ixia_mgmt_ip, get_neigh_ixia_card,\
+     get_neigh_ixia_port, IxiaFanoutManager
+
+import time
+
+def create_ipv4_traffic_end_points (
+    src_start_port,
+    src_port_count,
+    src_first_route_index,
+    src_route_count,
+    dst_start_port,
+    dst_port_count,
+    dst_first_route_index,
+    dst_route_count) :
+
+    src = [{'arg1': '/api/v1/sessions/1/ixnetwork/topology/1/deviceGroup/1/ethernet/1/ipv4/1',
+             'arg2': src_start_port, 
+             'arg3': src_port_count, 
+             'arg4': src_first_route_index, 
+             'arg5': dst_route_count}
+    ]
+
+    dst = [{'arg1': '/api/v1/sessions/1/ixnetwork/topology/1/deviceGroup/1/ethernet/1/ipv4/1', 
+             'arg2': dst_start_port, 
+             'arg3': dst_port_count, 
+             'arg4': dst_first_route_index,
+             'arg5': dst_route_count}
+    ]
+
+    return (src, dst)
+
+
+def test_testbed(testbed, conn_graph_facts, duthost, fanout_graph_facts,
```

Comment:
This was one of my earlier comments. This method contains all the setup, test steps. Ideally all the setup/teardown should be in a method of its own, possibly a fixture. This test method should only contain the test execution/verify something like the following

@pytest.fixture(scope='relevant scope')
def setup_teardown():
... setup steps
yield
... teardown steps

def test_testbed(setup_teardown):
... test execution/verify

Also I don't see any cleanup as part of this test.

Reply:
I have taken care of this issue. changed the fixture ixia_api_server_session